ios/engine/KMEI/KeymanEngine/Classes/Manager.swift

```diff
@@ -307,29 +315,20 @@ public class Manager: NSObject, HTTPDownloadDelegate, UIGestureRecognizerDelegat
   ///   - addLexicalModel()
   /// - Returns: Whether the lexical model was set successfully
   //TODO: this method appears unused, should we remove it?
-  public func setLexicalModel(withFullID fullID: FullLexicalModelID) -> Bool {
+  public func registerLexicalModel(withFullID fullID: FullLexicalModelID) -> Bool {
     if let lexicalModel = Storage.active.userDefaults.userLexicalModel(withFullID: fullID) {
-      return setLexicalModel(lexicalModel)
+      return registerLexicalModel(lexicalModel)
     }
     return false
   }
   
-  
-  /// Set the current lexical model.
-  ///
-  /// - Throws: error if the lexical model was unchanged
-  public func setLexicalModel(_ lm: InstallableLexicalModel) -> Bool {
-    if lm.fullID == currentLexicalModelID {
-      log.info("Lexical model unchanged: \(lm.fullID)")
-      return false
-      // throw LexicalModelError.unchanged
-    }
-    
-    log.info("Setting language: \(lm.fullID)")
+  /// Registers a lexical model with KMW.
+  public func registerLexicalModel(_ lm: InstallableLexicalModel) -> Bool {
+    log.info("Setting lexical model: \(lm.fullID)")
     
     currentLexicalModelID = lm.fullID
     
-    inputViewController.setLexicalModel(lm)
+    inputViewController.registerLexicalModel(lm)
```

Comment:
This looks strange. Why are you doing this?

Reply:
kmwembedded.ts:

```
  /**
   * Register a lexical model
   * 
   * @param {com.keyman.text.prediction.ModelSpec} model  Spec of the lexical model
   */
  keymanweb['registerModel']=function(model: com.keyman.text.prediction.ModelSpec) {
    keymanweb.modelManager.register(model);
  };
```

It more directly parallels the nomenclature in KMW and better reflects how lexical models are managed therein.